Change the 'is-ready' postMessage pairs to 'ready:request' and 'ready:response' to be more explicit

canova opened the pull request:
Edit after updating the PR:
I have changed the postMessage names directly now to `'ready:request'` and `'ready:response'` to be more explicit.

Before update:
This changes the `is-ready` message handling that comes from `postMessage`. I needed this for a Chrome extension I was working on. But in their extension API you have to be in the same `window` while sending this message. That's why when `is-ready` returns a message with the same name to itself, it creates an infinite loop. I changed the name from `is-ready` to `ready` when notifying the profiler is ready.

@gregtatum I wanted to get your feedback since you implemented this feature initially for your work. I would prefer to always send `ready` instead of `is-ready` but that would probably break your use case. So that's why I went with a non-breaking usage where I check if the windows are the same and if so, send a message with a different name. Let me know what you think!

julienw commented:
Ideally i'd use `ready:request` and `ready:response`, something like that. But indeed it is a breaking update, it would be good to know if this can be easily changed on your side @gregtatum :-)

gregtatum commented:
I've been afk for personal stuff these last few weeks, I can look at this soon, but feel free to merge without me.

gregtatum commented:
Please change it to what makes most sense for the cleanest implementation, and I can easily update the message name when you merge. Mine's easy to change.

canova commented:
Thanks for the feedback @gregtatum! And thanks for the suggestion @julienw on the message names. I updated the PR and changed the `is-ready` pairs to `ready:request` and `ready:response`.

Please let me know what you think!

gregtatum reviewed (approved):
LGTM!

julienw reviewed (approved):
lgtm as well!

canova commented:
Note that I'm deploying in a bit today. So we would need to update the places where we use this message.
